tests/v2/validation/provisioning/rke2/encryption_key_rotation_test.go

```diff
@@ -0,0 +1,270 @@
+	suite.Suite
+	session     *session.Session
+	client      *rancher.Client
+	config      *provisioning.Config
+	clusterName string
+	namespace   string
+}
+
+const scale = 10000
+
+var phases = []rkev1.RotateEncryptionKeysPhase{
+	rkev1.RotateEncryptionKeysPhasePrepare,
+	rkev1.RotateEncryptionKeysPhasePostPrepareRestart,
+	rkev1.RotateEncryptionKeysPhaseRotate,
+	rkev1.RotateEncryptionKeysPhasePostRotateRestart,
+	rkev1.RotateEncryptionKeysPhaseReencrypt,
+	rkev1.RotateEncryptionKeysPhasePostReencryptRestart,
+	rkev1.RotateEncryptionKeysPhaseDone,
+}
+
+func (r *RKE2EncryptionKeyRotationTestSuite) TearDownSuite() {
+	r.session.Cleanup()
+}
+
+func (r *RKE2EncryptionKeyRotationTestSuite) SetupSuite() {
+	testSession := session.NewSession(r.T())
+	r.session = testSession
+
+	r.config = new(provisioning.Config)
+	config.LoadConfig(provisioning.ConfigurationFileKey, r.config)
+
+	client, err := rancher.NewClient("", testSession)
+	require.NoError(r.T(), err)
+
+	r.client = client
+
+	r.clusterName = r.client.RancherConfig.ClusterName
+	r.namespace = r.client.RancherConfig.ClusterName
+}
+
+func (r *RKE2EncryptionKeyRotationTestSuite) TestEncryptionKeyRotationFreshCluster(provider Provider, kubeVersion string, nodesAndRoles []machinepools.NodeRoles, credential *cloudcredentials.CloudCredential) {
+	name := fmt.Sprintf("Provider_%s/Kubernetes_Version_%s/Nodes_%v", provider.Name, kubeVersion, nodesAndRoles)
+	r.Run(name, func() {
+		r.Run("initial", func() {
+			testSession := session.NewSession(r.T())
+			defer testSession.Cleanup()
+
+			testSessionClient, err := r.client.WithSession(testSession)
+			require.NoError(r.T(), err)
+
+			clusterName := namegen.AppendRandomString(fmt.Sprintf("%s-%s", r.clusterName, provider.Name))
+			generatedPoolName := fmt.Sprintf("nc-%s-pool1-", clusterName)
+			machinePoolConfig := provider.MachinePoolFunc(generatedPoolName, namespace)
+
+			machineConfigResp, err := testSessionClient.Steve.SteveType(provider.MachineConfigPoolResourceSteveType).Create(machinePoolConfig)
+			require.NoError(r.T(), err)
+
+			machinePools := machinepools.RKEMachinePoolSetup(nodesAndRoles, machineConfigResp)
+
+			cluster := clusters.NewK3SRKE2ClusterConfig(clusterName, namespace, "calico", credential.ID, kubeVersion, machinePools)
+
+			if strings.Contains(kubeVersion, "k3s") {
+				cluster.Spec.RKEConfig.MachineGlobalConfig.Data["secrets-encryption"] = true
+			}
+
+			clusterResp, err := clusters.CreateK3SRKE2Cluster(testSessionClient, cluster)
+			require.NoError(r.T(), err)
+
+			kubeProvisioningClient, err := r.client.GetKubeAPIProvisioningClient()
+			require.NoError(r.T(), err)
+
+			result, err := kubeProvisioningClient.Clusters(namespace).Watch(context.TODO(), metav1.ListOptions{
+				FieldSelector:  "metadata.name=" + cluster.ObjectMeta.Name,
+				TimeoutSeconds: &defaults.WatchTimeoutSeconds,
+			})
+			require.NoError(r.T(), err)
+
+			checkFunc := clusters.IsProvisioningClusterReady
```

Comment:
Nit: You're creating this variable but it's only used in one place, which happens to be on the next line.

Reply:
Just copying the logic here: https://github.com/rancher/rancher/blob/10f2cc093c4864a694981338c034acfdd20b6548/tests/v2/validation/provisioning/rke2/custom_cluster.go#L81-L82

Now that I think about it, your comment is probably the more correct way to do it.